src/validators/callout-component.validator.ts

```diff
@@ -103,4 +106,46 @@ export const calloutComponentValidator = (
   }
 
   return true;
-};
+}
+
+export function calloutComponentValidator(
+  schema: CalloutComponentSchema,
+  answer:
+    | CalloutResponseAnswer
+    | CalloutResponseAnswer[]
+    | undefined,
+): boolean;
+
+export function calloutComponentValidator(
+  schema: CalloutComponentNestableSchema,
+  answer: CalloutResponseAnswersNestable,
+): boolean;
+
+export function calloutComponentValidator(
+  schema: CalloutComponentSchema,
+  answer:
+    | CalloutResponseAnswer
+    | CalloutResponseAnswer[]
+    | undefined
+    | CalloutResponseAnswersNestable,
```

Comment:
I've never quite understood the `CalloutResponseAnswersNestable` type, I think this might be a misunderstanding of the data schema.

Nested components aren't nested in the answers object, everything is flattened. i.e. for the following component schema:

```json
{
  "key": "mypanel",
  "type": "panel",
  "components": [{
    "key": "mytext",
    "type": "textfield"
  }, {
    "key": "mynumber",
    "type": "number"
  }]
}
```

The main answers object would still look like this (not nested):
```json
{
  "mytext": "My text answer",
  "mynumber": 123
}
```

So I think you don't need that type at all which then makes this method simpler

Reply:
That is also what the type represents:

```ts
/**
 * Answers are grouped by component key: `{[componentKey]: answer | answer[]}`
 */
export type CalloutResponseAnswersNestable = Record<
  string,
  CalloutResponseAnswer | CalloutResponseAnswer[] | undefined
>;
```

Reply:
Maybe `CalloutResponseAnswersByComponentKey` would be a better name?

Reply:
I find this name `CalloutResponseAnswers` confusing because a single callout schema can have multiple answers when activated. 

Is the comment right for `CalloutResponseAnswersSlide`:

```ts
/**
 * Answers are grouped by slide key: `{[slideId]: {[componentKey]: answer | answer[]}}`
 */
export type CalloutResponseAnswersSlide = Record<
  string,
  CalloutResponseAnswersNestable | undefined
>;
```

Reply:
But since I can't think of a better name, I change it back to `CalloutResponseAnswers`